homeassistant/components/tile/device_tracker.py

```diff
@@ -1,140 +1,129 @@
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
 
-from homeassistant.components.device_tracker import PLATFORM_SCHEMA
-from homeassistant.const import CONF_MONITORED_VARIABLES, CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import aiohttp_client, config_validation as cv
-from homeassistant.helpers.event import async_track_time_interval
-from homeassistant.util import slugify
-from homeassistant.util.json import load_json, save_json
+from . import DATA_COORDINATOR, DOMAIN, TileEntity
 
 _LOGGER = logging.getLogger(__name__)
-CLIENT_UUID_CONFIG_FILE = ".tile.conf"
-DEVICE_TYPES = ["PHONE", "TILE"]
 
 ATTR_ALTITUDE = "altitude"
 ATTR_CONNECTION_STATE = "connection_state"
 ATTR_IS_DEAD = "is_dead"
 ATTR_IS_LOST = "is_lost"
 ATTR_RING_STATE = "ring_state"
 ATTR_VOIP_STATE = "voip_state"
-ATTR_TILE_ID = "tile_identifier"
 ATTR_TILE_NAME = "tile_name"
 
-CONF_SHOW_INACTIVE = "show_inactive"
 
-DEFAULT_ICON = "mdi:view-grid"
-DEFAULT_SCAN_INTERVAL = timedelta(minutes=2)
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Tile device trackers."""
+    coordinator = hass.data[DOMAIN][DATA_COORDINATOR][config_entry.entry_id]
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_SHOW_INACTIVE, default=False): cv.boolean,
-        vol.Optional(CONF_MONITORED_VARIABLES, default=DEVICE_TYPES): vol.All(
-            cv.ensure_list, [vol.In(DEVICE_TYPES)]
-        ),
-    }
-)
+    async_add_entities(
+        [
+            TileDeviceTracker(coordinator, tile_uuid, tile)
+            for tile_uuid, tile in coordinator.data.items()
+        ],
+        True,
+    )
 
 
 async def async_setup_scanner(hass, config, async_see, discovery_info=None):
-    """Validate the configuration and return a Tile scanner."""
-    websession = aiohttp_client.async_get_clientsession(hass)
-
-    config_file = hass.config.path(
-        ".{}{}".format(slugify(config[CONF_USERNAME]), CLIENT_UUID_CONFIG_FILE)
-    )
-    config_data = await hass.async_add_job(load_json, config_file)
-    if config_data:
-        client = await async_login(
-            config[CONF_USERNAME],
-            config[CONF_PASSWORD],
-            websession,
-            client_uuid=config_data["client_uuid"],
-        )
-    else:
-        client = await async_login(
-            config[CONF_USERNAME], config[CONF_PASSWORD], websession
+    """Detect a legacy configuration and import it."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_USERNAME: config[CONF_USERNAME],
+                CONF_PASSWORD: config[CONF_PASSWORD],
+            },
```

Comment:
Isn't the import still needed ?

Reply:
Sorry, I got confused by your comment above on coverage. I see now.